llama-index-integrations/readers/llama-index-readers-file/llama_index/readers/file/upstage/base.py

```diff
@@ -0,0 +1,314 @@
+import io
+import os
+from pathlib import Path
+from typing import Dict, List, Union
+
+import fitz
+import requests
+from llama_index.core.readers.base import BaseReader
+from llama_index.core.schema import Document
+from llama_index.legacy.llms.generic_utils import get_from_param_or_env
```

Comment:
Ah I missed this. Should be `from llama_index.core.base.llms.generic_utils import get_from_param_or_env`